packages/react-vega/src/Vega.jsx

```diff
@@ -114,15 +116,18 @@ class Vega extends React.Component {
     this.clearView();
   }
 
-  createView(spec) {
+  async createView(spec) {
     if (spec) {
       const { props } = this;
       // Parse the vega spec and create the view
       try {
-        const runtime = vega.parse(spec);
-        const view = new vega.View(runtime).initialize(this.element);
-
-        // Attach listeners onto the signals
+        let view = {};
+        try {
+          result = await vegaEmbed(this.element, spec);
+          ({ view } = result);
+        } catch (error) {
+          throw new Error('Something went wrong parsing Vega specs');
```

Comment:
What is the benefit of catching to rethrow and ignoring the original error message?

Reply:
I agree. Let's just remove this and let people take care of handing errors.